apps/organizations/apps.py

```diff
@@ -19,21 +19,29 @@
 
 
 @disable_for_loaddata
-def organization_invitation_created(sender, instance, created, **kwargs):
-    """Signal handler for when a new OrganizationInvitation object is created"""
-    if created:
-        invitation = instance
-
-        if not settings.TEST and htk_setting('HTK_SLACK_NOTIFICATIONS_ENABLED'):
+def organization_invitation_updated(sender, instance, created, **kwargs):
+    """Signal handler for when a new OrganizationInvitation object is created or updated"""
+    invitation = instance
+    if not settings.TEST and htk_setting('HTK_SLACK_NOTIFICATIONS_ENABLED'):
+        from htk.utils.notifications import slack_notify
+
+        if created:
+            msg = invitation.build_notification_message__created()
+        else:
+            INVITATION_ACCEPTANCE_MESSAGE_BUILDERS = {
+                True: invitation.build_notification_message__accepted,
+                False: invitation.build_notification_message__declined,
+                None: invitation.build_notification_message__resent,
+            }
+            msg_builder = INVITATION_ACCEPTANCE_MESSAGE_BUILDERS.get(invitation.accepted)
+
+        if msg_builder is not None:
+            msg = msg_builder()
             try:
-                from htk.utils.notifications import slack_notify
-
-                msg = invitation.build_notification_message__created()
                 slack_notify(msg)
```

Comment:
Unless you ensure that all of the functions will always return a message, you will also need to check `if msg is not None` because it looks like there is a case where you don't return a message.